web/static/js/components/group_label_input.jsx

```diff
@@ -7,23 +7,63 @@ import styles from "./css_modules/group_label_input.css"
 // exceed the 1/4th column width on desktop, even with the widest possible characters
 const MAX_LENGTH_OF_GROUP_NAME = 20
 
-const GroupLabelInput = ({ groupWithAssociatedIdeasAndVotes, actions }) => (
-  <div className="ui fluid input">
-    <input
-      type="text"
-      className={styles.textInput}
-      placeholder="Add a group label"
-      defaultValue={groupWithAssociatedIdeasAndVotes.label}
-      maxLength={MAX_LENGTH_OF_GROUP_NAME}
-      onBlur={e => {
-        actions.submitGroupLabelChanges(groupWithAssociatedIdeasAndVotes, e.target.value)
-      }}
-    />
-    <div className="instruction">
-      <span className="keyboard-key">tab</span> to submit
-    </div>
-  </div>
-)
+class GroupLabelInput extends Component {
+  state = (() => {
+    const { groupWithAssociatedIdeasAndVotes } = this.props
+    const { label } = groupWithAssociatedIdeasAndVotes
+    return {
+      oldLabel: label,
+      showCheckmarkForLabelId: null,
+    }
+  })();
+
+  static getDerivedStateFromProps(props, state) {
+    const { oldLabel } = state
+    const { groupWithAssociatedIdeasAndVotes } = props
+    const newLabel = groupWithAssociatedIdeasAndVotes.label
+    if (oldLabel !== newLabel) {
+      return ({
+        oldLabel: newLabel,
+        showCheckmarkForLabelId: groupWithAssociatedIdeasAndVotes.id })
+    }
+    return null
+  }
+
+  labelUpdatedCheckmark = () => {
+    const classes = ["check icon"]
+    classes.push(styles.updateSucceededCheckmark)
+    setTimeout(() => {
+      this.setState({ showCheckmarkForLabelId: null })
+    }, 2000)
+    return <div className="ui"><i className={classes.join(" ")} /></div>
+  }
+
+  render() {
+    const { groupWithAssociatedIdeasAndVotes, actions } = this.props
+    const { showCheckmarkForLabelId } = this.state
+
+    return (
+      <div className="ui fluid input">
+        <input
+          type="text"
+          className={styles.textInput}
+          placeholder="Add a group label"
+          defaultValue={groupWithAssociatedIdeasAndVotes.label}
+          maxLength={MAX_LENGTH_OF_GROUP_NAME}
+          onBlur={e => {
+            actions.submitGroupLabelChanges(groupWithAssociatedIdeasAndVotes, e.target.value)
+          }}
+        />
+        <div className="instruction">
+          <span className="keyboard-key">tab</span> to submit
+        </div>
+        {showCheckmarkForLabelId === groupWithAssociatedIdeasAndVotes.id
```

Comment:
per my note above, the group id will be unique to the instance of this component, so i think we can simplify the state attribute to a boolean at `showCheckmarkForLabel`, and simplify the render code to a short-circuit, like so: {`showCheckmarkForLabel && this.labelUpdatedCheckmark()`}

Reply:
We initially added the label id check since we found that all components were being updated when only one's label was modified.  However, with the new code, this no longer happens.  Thanks.  Modifying as per suggestion.